atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/CommitTsLoader.java

```diff
@@ -49,12 +52,22 @@ private CommitTsLoader(TLongLongMap commitTsByStartTs, TransactionService transa
     public static CommitTsLoader create(TransactionService transactionService, TLongSet startTssToWarmingCache) {
         TLongLongMap cache = new TLongLongHashMap();
         if (!startTssToWarmingCache.isEmpty()) {
-            // Ideally TransactionService should work with primitive collections to avoid GC overhead..
-            cache.putAll(transactionService.get(TDecorators.wrap(startTssToWarmingCache)));
+            warmCache(transactionService, startTssToWarmingCache, cache);
```

Comment:
Curious why we don't consider warming the cache async, especially if we may be loading a million pairs during the warming phase. Especially since the default cache size is one million, so loading any more than that is going to end up being useless.

Reply:
(Not necessarily in the scope of this PR - we should resolve the P0 as a higher priority)

Reply:
As @j-baker noted, sweep is also be a bad user of the cache, since it pollutes it with timestamps that are likely to be uncorrelated with the timestamps read by live requests.

Reply:
Checked the above and we _only_ use the cache for sweep, not for actual transactions, where I'd guess it may be more useful...

Reply:
Fun. 

Transactions use a `TimestampCache` which is by default limited to 1 million entries. It's a dumb cache class with no logic for loading timestamps or rolling back transactions.

Sweep uses a different class `CommitTsLoader` instead, which doesn't really act as a cache. Each `CommitTsLoader` only lives as long as it takes to complete a call to `SweepableCellFilter.getCellsToSweep`, so we only benefit from the cache if, if I understand correctly, multiple cells in a single row were written with the same timestamp, which certainly should happen. That said, we are throwing away the cache completely as soon as we move on to the next batch. Background sweep should use a different cache from the transactions, since the timestamps they load are very likely to be different, but we might benefit from combining the classes used and not blowing away the cache so frequently.

Also, `CommitTsLoader` has no eviction and targeted sweep keeps a long lived one around... filed https://github.com/palantir/atlasdb/issues/3246.

Sorry for all the confusion above, it was pretty surprising to me that we have two separate commit timestamp cache implementations :(

Reply:
Correct, yeah these are two distinct caches 😞

Yep, rather confusingly we also throw away the `CommitTsLoader`. (Technically the cache gives you benefits if your rows are small and rows that are nearby share timestamps too)

I wrote this PR as a fairly quick hack to defuse the P0, but I'd want to unify the caches at some point